src/pyird/image/aptrace.py

```diff
@@ -37,17 +52,28 @@ def set_aperture(dat,cutrow,nap):
     """
     # Search peaks in the cross section at cutrow
     # cutrow is selected so that the number of peaks and nap match
+    cutrow_min=500
+    cutrow_max=1550
     peakind_cut = []
-    while len(peakind_cut) != nap:
-        onerow_masked,med = cross_section(dat,cutrow)
-        peakind_cut, _ = find_peaks(onerow_masked,height=15*np.median(med))
-        print('cross-section: row ',cutrow)
+    cutrow_lim = True
+    while ((len(peakind_cut)!=nap) or prange) and cutrow_lim:
+        onerow_masked,peakind_cut = cross_section(dat,cutrow,nap)
+        #print('cross-section: row ',cutrow)
+        if nap<50: # h band
+            prange = (peakind_cut[-1]>1500) or (peakind_cut[1]>40)
+        elif nap>100: # yj band
+            prange = (peakind_cut[0]<250) or (peakind_cut[-2]<2000)
+        cutrow_lim = (cutrow>cutrow_min) and (cutrow<cutrow_max)
         cutrow += 1
-
+    if not cutrow_lim:
+        print('Error: %d apertures could not be found.'%(nap))
+        print('Please change "cutrow" or "nap" or confirm the orientation of the image.')
+        sys.exit()
```

Comment:
Considering the case one runs the code in Jupyter notebook, `return` is better?
If you agree, remove `import sys` too.

Reply:
I agree. I will change.